api/agent/call.go

```diff
@@ -260,6 +262,9 @@ func (a *agent) GetCall(opts ...CallOpt) (Call, error) {
 type call struct {
 	*models.Call
 
+	app   *models.App
+	route *models.Route
```

Comment:
pretty similar to the config stuff: https://github.com/fnproject/fn/blob/master/api/agent/call.go#L115-L122 - we could take one out of that book and hang it off `*models.Call`, then merge the app and route annotations into one `Annotations` bucket. notably, for async, the config goes in the json blob message to the mq (a json of `models.Call`) so that we can later use that config after we pull the message off without having to rebuild it. annotations seems to be analogous to this, and for some kind of cohesion it makes sense for now (we could decide to chop up `*models.Call` some day). thoughts?

I do think I prefer this to #878 for this use case

Reply:
Also for context there are objections to extending the models.Call object since that will be persisted with every call.